ui/helpers/utils/tags.ts

```diff
@@ -0,0 +1,41 @@
+// TODO: Remove restricted import
+// eslint-disable-next-line import/no-restricted-paths
```

Comment:
This is unfortunate. `getEnvironmentType` should always have been in `shared` I guess.

Typically in this scenario I'd ask that we move it rather than leaving this as a TODO, but given that we intend to cherry-pick this into v12.5.0, moving this in a follow-up PR seems better (to minimize conflicts).